forge-game/src/main/java/forge/game/staticability/StaticAbilityPanharmonicon.java

```diff
@@ -157,6 +156,17 @@ public static boolean applyPanharmoniconAbility(final StaticAbility stAb, final
                 return false;
             }
         } else if (trigMode.equals(TriggerType.DamageDone) || trigMode.equals(TriggerType.DamageDoneOnce)) {
+            if (stAb.hasParam("ValidSource")) {
+                if (trigMode.equals(TriggerType.DamageDone) && 
+                    !stAb.matchesValidParam("ValidSource", runParams.get(AbilityKey.DamageSource))) return false;
+                else if (trigMode.equals(TriggerType.DamageDoneOnce) && 
+                    (int) runParams.get(AbilityKey.DamageAmount) <= 0) return false;
+                //DamageAmount is filtered by ValidSource in TriggerDamageDoneOnce
```

Comment:
hmn, I have to say I don't quite understand how this comment lets us avoid filtering for the Source?

Reply:
also isn't that the wrong mode anyway compared to Wayta?

Reply:
@Hanmac
I believe DamageAll will also need to be included for this one?

Reply:
yeah, probably DamageAll too

Reply:
Should [Wayta, Trainer Prodigy](https://scryfall.com/card/lcc/7/wayta-trainer-prodigy) also have DamageAll ?

Reply:
- I meant DamageDealtOnce instead, make sure you're not mixing up "being dealt" and "dealing"

- it looks like you're trying to use triggeringObjects? that's not how it works (they're not available)

- DamageAll needs a lot more checking, it's a bit tricky to explain but I'll try and give you an example:

Consider a combat scenario in which one of your two attacking creatures gets a warrior blocker where the damage it receives is redirected to the defending player

The final damage event will look like this:
attacker A -> 1+ dmg to Opp
blocker B -> 1+ dmg to Opp

now look at _Mindblade Render_:
> Whenever your opponents are dealt combat damage, if any of that damage was dealt by a Warrior, you draw a card and you lose 1 life.

does he trigger? yes
and does Felix double it? **no!**

because it wasn't actually **your** warrior that caused it

-> you can't check the params in isolation but a single entry in the DamageMap needs to match all of them together

you can also check the work I did for ChangesZoneAll as guideline (luckily you don't need to worry about DisableTriggers for now :P)

- Chances are you _could_ ignore it on Wayta for now because it can't match anyway

Reply:
Thank you for all this – looks like you're on Discord right now, I might ping you

Reply:
I believe I fixed it so now it checks based on the potential triggeringObjects too
Feel free to test
Imo we don't need to worry about ExcessDamage yet, since it seems impossible to match right now 🤔

Reply:
are the Damage Trigger ready yet?

Reply:
I did not have time for testing yet

Reply:
ok, looks like it behaves the way I want